aptos-move/framework/aptos-framework/sources/Token.move

```diff
@@ -29,28 +40,99 @@ module AptosFramework::Token {
     // Core data structures for holding tokens
     //
 
-    /// Represents ownership of a the data associated with this Token
     struct Token has store {
         id: TokenId,
+        // the amount of tokens
         value: u64,
     }
 
+    /// global unique identifier of a token
+    struct TokenId has store, copy, drop {
+        // the id to the common token data shared by token with different serial number
+        token_data_id: TokenDataId,
+        // the serial number, every mint of this token_data_id will generate a new serial number
+        serial_number: u64
+    }
 
-    /// Represents a unique identity for the token
-    struct TokenId has copy, drop, store {
+    /// global unique identifier of tokendata
+    struct TokenDataId has copy, drop, store {
         // The creator of this token
         creator: address,
         // The collection or set of related tokens within the creator's account
-        collection: ASCII::String,
-        // Unique name within a collection within the creator's account
-        name: ASCII::String,
+        collection: String,
+        // the name of this token
+        name: String,
+    }
+
+    /// The royalty of a token
+    struct Royalty has copy, drop, store {
+        royalty_points_nominator: u64,
+        royalty_points_denominator: u64,
+        // if the token is jointly owned by multiple creators,
+        // the group of creators should create a shared account.
+        // the payee_address will be the shared account address.
+        payee_address: address,
+    }
+
+    /// This config specifies which fields in the Collection are mutable
+    struct CollectionMutabilityConfig has copy, store, drop {
+        // control if description is mutable
+        description: bool,
+        // control if uri is mutable
+        uri: bool,
+        // control if collection maxium is mutable
+        maximum: bool
+    }
+
+    /// This config specifies which fields in the TokenData are mutable
+    struct TokenMutabilityConfig has copy, store, drop {
+        // control if the token maximum is mutable
+        maximum: bool,
+        // control if the token uri is mutable
+        uri: bool,
+        // control if the token royalty is mutable
+        royalty: bool,
+        // control if the token description is mutable
+        description: bool,
+        // control if the default token property list are mutable
+        default_property_list: bool,
+        // control if token with different serial number can have different value
+        property_values: bool
+    }
```

Comment:
I might order the structs in a way that makes it easier to understand where we're going:

* Collection
* CollectionMutability
* TokenData
* TokenDataMutability
* Token
* TokenId

Reply:
updated